src/vulkan-renderer/wrapper/window.cpp

```diff
@@ -12,19 +12,21 @@ Window::Window(const std::string &title, const std::uint32_t width, const std::u
     assert(!title.empty());
 
     glfwWindowHint(GLFW_CLIENT_API, GLFW_NO_API);
-    glfwWindowHint(GLFW_VISIBLE, visible);
-    glfwWindowHint(GLFW_RESIZABLE, resizable);
+    glfwWindowHint(GLFW_VISIBLE, visible ? GLFW_TRUE : GLFW_FALSE);
+    glfwWindowHint(GLFW_RESIZABLE, resizable ? GLFW_TRUE : GLFW_FALSE);
 
     spdlog::debug("Creating window.");
 
-    m_window = glfwCreateWindow(width, height, title.c_str(), nullptr, nullptr);
+    m_window = glfwCreateWindow(static_cast<int>(width), static_cast<int>(height), title.c_str(), nullptr, nullptr);
 
-    if (!m_window) {
+    if (m_window == nullptr) {
```

Comment:
huh? Wasn't this considered bad style?

Reply:
Otherwise it would be an implicit conversion, for which we need a `static_cast` to be absolutely correct.

Reply:
ufff. But but [C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions).....

Reply:
I personally think the explicit check is better, since just doing `if (!m_window)` is technically doing the same as `if (m_window != 0)` instead of `if (m_window != nullptr)`. But if we are following the C++ core guidelines exactly then...

Reply:
Let's keep it for now?